### PR TITLE
feat: make the prefix in the solution optional across resource names

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Allows a consumer to provide either:
* prefix + name of a resource / group
* name of a resource / group

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/account-infrastructure-base/variables.tf [hunk 1]
```diff
@@ -18,6 +18,7 @@ variable "region" {
 variable "prefix" {
   type        = string
   description = "Prefix to append to all resources created by this solution."
```
Comment (ocofaigh) [inline solutions/account-infrastructure-base/variables.tf:20]: Can you add `(Optional)` to the start of this descrption?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@MatthewLemmond please update all variable descriptions that would not be impacted if a prefix value was passed. If you take a look at https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/pull/99 for example, you can see we added the text `If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'` to all of them.

**Comment by MatthewLemmond** [comment]
> @MatthewLemmond can you confirm if did a test not passing prefix to confirm this works as expected?

[here](https://github.com/terraform-ibm-modules/terraform-ibm-account-infrastructure-base/actions/runs/8805467544/job/24168251303) is a run that didn't have a prefix provided and it successfully created the resource group without a prefix (failed due to name conflict on the bucket) so this serves as a proof of concept that it's working without the prefix being passed in

File: solutions/account-infrastructure-base/variables.tf [hunk 2]
```diff
@@ -56,43 +57,43 @@ variable "provision_atracker_cos" {
...
   type        = string
-  description = "The name of the observability resource group to create. Required if `var.provision_atracker_cos` is true and `var.existing_cos_resource_group_name` is not provided."
+  description = "The name of the observability resource group to create. Required if `var.provision_atracker_cos` is true and `var.existing_cos_resource_group_name` is not provided. If `var.prefix` is provided then it will be prefixed on the name in the format of `<var.prefix>-<var.observability_resource_group_name>`."
   default     = null
 }
 
 variable "management_resource_group_name" {
   type        = string
-  description = "The name of the management resource group to create."
+  description = "The name of the management resource group to create. If `var.prefix` is provided then it will be prefixed on the name in the format of `<var.prefix>-<var.management_resource_group_name>`."
```
[the remark this answers is not in the document]
Comment (MatthewLemmond) [reply solutions/account-infrastructure-base/variables.tf]: removed the 'var.' when referencing variables from the descriptions

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**